### PR TITLE
Docs: azurerm_resource_group_template_deployment show passing variables

### PR DESCRIPTION
**Pull request by lawrencegripper**
Tweak to the docs to demonstrate how to pass TF vars to the ARM template.

### CODE REVIEW

File: website/docs/r/resource_group_template_deployment.html.markdown [hunk 1]
```diff
@@ -15,31 +15,47 @@ Manages a Resource Group Template Deployment.
 ## Example Usage
 
 ```hcl
+locals {
+  vnet_name = "example-vnet"
+}
+
 resource "azurerm_resource_group_template_deployment" "example" {
-  name                = "example-deploy"
+  name                = "lgtest1"
```
Comment (katbyte) [inline website/docs/r/resource_group_template_deployment.html.markdown:23]: could we revert this
```suggestion
  name                = "example-deploy"
```

**Review by katbyte — APPROVED** [review]
aside from 1 comment this LGTM 👍  htanks @lawrencegripper

File: website/docs/r/resource_group_template_deployment.html.markdown [hunk 2]
```diff
@@ -15,31 +15,47 @@ Manages a Resource Group Template Deployment.
...
 resource "azurerm_resource_group_template_deployment" "example" {
   name                = "example-deploy"
   resource_group_name = "example-group"
   deployment_mode     = "Complete"
+  parameters_content  = jsonencode({
+    "vnetName" = {
+      value = local.vnet_name
+    }
+  })
   template_content    = <<TEMPLATE
```
Comment (katbyte) [inline website/docs/r/resource_group_template_deployment.html.markdown:31]:
```suggestion
  template_content = <<TEMPLATE
```

File: website/docs/r/resource_group_template_deployment.html.markdown [hunk 3]
```diff
@@ -15,31 +15,47 @@ Manages a Resource Group Template Deployment.
...
-      }
-    }
-  ]
+    },
+    "variables": {},
+    "resources": [
+        {
+            "type": "Microsoft.Network/virtualNetworks",
+            "apiVersion": "2020-05-01",
+            "name": "acctest-network",
```
Comment (tombuildsstuff) [inline website/docs/r/resource_group_template_deployment.html.markdown:48]: for the purposes of this example we should probably use this too, e.g.

```suggestion
            "name": "[variables("vnetName")]",
```

File: website/docs/r/resource_group_template_deployment.html.markdown [hunk 4]
```diff
@@ -66,7 +82,7 @@ The following arguments are supported:
 
 * `debug_level` - (Optional) The Debug Level which should be used for this Resource Group Template Deployment. Possible values are `none`, `requestContent`, `responseContent` and `requestContent, responseContent`.
 
-* `parameters_content` - (Optional) The contents of the ARM Template parameters file - containing a JSON list of parameters.
+* `parameters_content` - (Optional) The contents of the ARM Template parameters file - containing a JSON list of parameters. See example for how to pass in Terraform Variable to an ARM parameter.
```
Comment (tombuildsstuff) [inline website/docs/r/resource_group_template_deployment.html.markdown:85]: Let's make this a blue box:

```suggestion
* `parameters_content` - (Optional) The contents of the ARM Template parameters file - containing a JSON list of parameters.

-> An example of how to pass Terraform variables into an ARM Template can be seen in the example.
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @lawrencegripper 

Thanks for pushing those changes

Taking a look through I think it'd be good to demonstrate using the new variable in this example, but if we can fix that up then this should otherwise be good to go 👍 

Thanks!

File: website/docs/r/resource_group_template_deployment.html.markdown (hunk 3 above)
Comment (lawrencegripper) [reply website/docs/r/resource_group_template_deployment.html.markdown]: Agree, good call.

**Comment by lawrencegripper** [comment]
Thanks for taking a look, suggestions look good to me. Have accepted them all.

**Comment by lawrencegripper** [comment]
I've added in additional example of how to use ARM outputs from the template in TF too.

**Review by katbyte — APPROVED** [review]
LGTM 👍

**Pull request merged**